datalad/support/gitrepo.py

```diff
@@ -64,10 +64,58 @@
 # Didn't work as expected on a first try. Probably there is a neatier way to
 # log Exceptions from git commands.
 
-# TODO: Check whether it makes sense to unify passing of options in a way
-# similar to paths. See options_decorator in annexrepo.py
-# Note: GitPython is doing something similar already with **kwargs.
-# TODO: Figure this out in detail.
+
+@optional_args
+def kwargs_to_options(func, split_single_char_options=True,
+                      target_kw='options'):
+    """Decorator to provide convenient way to pass options to command calls.
+
+    Parameters
+    ----------
+    func: Callable
+        function to decorate
+    split_single_char_options: bool
+        whether or not to split key and value of single char keyword arguments
+        into two subsequent entries of the list
+    target_kw: str
+        keyword argument to pass the generated list of cmdline arguments to
+
+    Returns
+    -------
+    Callable
+    """
+
+    # TODO: don't overwrite options, but join
+
+    @wraps(func)
+    def newfunc(self, *args, **kwargs):
+        t_kwargs = dict()
+        t_kwargs[target_kw] = \
+            gitpy.Git().transform_kwargs(
+                        split_single_char_options=split_single_char_options,
+                        **kwargs)
+        return func(self, *args, **t_kwargs)
+    return newfunc
+
+
+def to_options(**kwargs):
+    """Transform keyword arguments into a list of cmdline options
+
+    Parameters
+    ----------
+    split_single_char_options: bool
+
+    kwargs:
+
+    Returns
+    -------
+    list
+    """
+    # TODO: borrow_docs!
+
+    split = kwargs.pop('split_single_char_options', True)
+    return gitpy.Git().transform_kwargs(split_single_char_options=split,
+                                        **kwargs)
```

Comment:
so what is the point of explicitly popping split_single_char_options?  or is it False by default in `transform_kwargs`?